src/EnergyPlus/DXCoils.cc

```diff
@@ -14966,10 +14965,15 @@ void GetFanIndexForTwoSpeedCoil(
                         break;
                     }
                     // these are specified in SimAirServingZones and need to be moved to a Data* file. UnitarySystem=19
-                } else if (state.dataAirSystemsData->PrimaryAirSystems(AirSysNum).Branch(BranchNum).Comp(CompNum).CompType_Num == UnitarySystem) {
-                    FoundBranch = BranchNum;
-                    FoundAirSysNum = AirSysNum;
-                    break;
+                } else if (state.dataAirSystemsData->PrimaryAirSystems(AirSysNum).Branch(BranchNum).Comp(CompNum).CompType_Num ==
+                           SimAirServingZones::UnitarySystemModel) {
+
+                    if (UtilityRoutines::SameString(state.dataAirSystemsData->PrimaryAirSystems(AirSysNum).Branch(BranchNum).Comp(CompNum).Name,
+                                                    state.dataDXCoils->DXCoil(CoolingCoilIndex).CoilSystemName)) {
+                        FoundBranch = BranchNum;
+                        FoundAirSysNum = AirSysNum;
+                        break;
+                    }
```

Comment:
Added check on name here to match what DXSystem did.

Reply:
@rraustad Note that this used to be hard coded to 19 inside this function. UnitarySystem hasn't been 19 for years. So, when this was enumed, it was set to `Furnace_UnitarySys_HeatOnly` which makes no sense inside a cooling coil function. So, basically this if has been wrong for-ever for UnitarySystem and this fixes it here.